src/Application.php

```diff
@@ -57,6 +57,24 @@ public function bootstrap()
         }
     }
 
+    /**
+     * Define the routes for an application.
+     *
+     * Use the provided RouteBuilder to define an application's routing, register scoped middlewares.
+     *
+     * @param \Cake\Routing\RouteBuilder $routes A route builder to add routes into.
+     * @return void
+     */
+    public function routes($routes)
+    {
+        // Register scoped middlewares for use in routes.php
+        $routes->registerMiddleware('csrf', new CsrfProtectionMiddleware([
+            'httpOnly' => true
+        ]));
```

Comment:
You can register this middleware in routes.php as well if you want everything in one place.

Reply:
Some middlewares requires `Application` as `$this`, some like `SecurityHeaderMiddleware` or `BodyParserMiddleware` have somewhat complex configuration or logic and I think `config/` should avoid complex logic, code/logic should be in `src/`.

`routes.php` should be for configuring routes, not middlewares.

Reply:
Ok. The parent method is just a require for `config/routes.php` and the `$routes` variable is available there too.